eng/scripts/automation/GenerateAndBuildLib.ps1

```diff
@@ -32,59 +42,94 @@ function Get-SwaggerInfo()
     }
     Write-Host "Cannot find swagger info"
-    Pop-Location
     exit 1
 }
 
-function Update-AutorestConfigFile() {
+function CreateOrUpdateAutorestConfigFile() {
     param (
         [string]$autorestFilePath,
+        [string]$namespace,
         [string]$inputfile = "",
-        [string]$readme = ""
+        [string]$readme = "",
+        [string]$autorestConfigYaml = ""
     )
-    if (Test-Path -Path $autorestFilePath) {
-        if ($readme -ne "") {
-            Write-Host "Updating autorest.md file to config required readme file."
+
+    if ((Test-Path -Path $autorestFilePath) -and (![String]::IsNullOrWhiteSpace((Get-Content -Path $autorestFilePath)))) {
+        if (($readme -ne "") -or ($inputfile -ne "")) {
             $requirRex = "require*:*";
-            $inputfileRex = "input-file *:*"
-            $requirefile = $readme + [Environment]::NewLine + "- " + $readme.Replace("readme.md", "readme.csharp.md")
-            if ((Get-Content $autorestFilePath | Select-String -Pattern $requirRex).Matches.Success) {
-                (Get-Content $autorestFilePath) -notmatch "- .*.md" |Out-File $autorestFilePath
-                (Get-Content $autorestFilePath) -notmatch $inputfileRex |Out-File $autorestFilePath
-                (Get-Content $autorestFilePath) -notmatch "- .*.json" |Out-File $autorestFilePath
-                (Get-Content $autorestFilePath) -replace $requirRex, ("require:" + [Environment]::NewLine + "- " + "$requirefile") | Set-Content $autorestFilePath
-            } elseif ((Get-Content $autorestFilePath | Select-String -Pattern $inputfileRex).Matches.Success) {
-                (Get-Content $autorestFilePath) -notmatch "- .*.json" |Out-File $autorestFilePath
-                $requirefile = $requirefile + [Environment]::NewLine + "csharp: true";
-                (Get-Content $autorestFilePath) -replace $inputfileRex, ("require:" + [Environment]::NewLine + "- " + "$requirefile") | Set-Content $autorestFilePath
+            $inputfileRex = "input-file*:*"
+            $fileContent = Get-Content -Path $autorestFilePath
+            # clear
+            $fileContent = $fileContent -notmatch $requirRex
+            $fileContent = $fileContent -notmatch "- .*.md"
+            $fileContent = $fileContent -notmatch "csharp: true"
+            $fileContent = $fileContent -notmatch $inputfileRex
+            $fileContent = $fileContent -notmatch "- .*.json"
+
+            $startNum = ($fileContent | Select-String -Pattern '```').LineNumber[0]
+            $configline = ""
+            if ($readme) {
+                Write-Host "Updating autorest.md file to config required readme file."
+                $requirefile = $readme
+                $configline = "require:" + [Environment]::NewLine + "- " + "$requirefile" + [Environment]::NewLine + "csharp: true"
+            } elseif ($inputfile) {
+                Write-Host "Updating autorest.md file to update input-file."
+                if ($inputfile.StartsWith('-')) {
```

Comment:
While I understand we support a list do we really need to use it when we generate our own file?

Reply:
yes. The script both support in swagger CI pipeline which will use readme as configure, and also it will support SDK-package-starter which may have input-file lists as input.